README.md

```diff
@@ -1,2 +1,95 @@
+    activate UPClientFoo owned thread / task
+
+    main thread->>main thread: ustreamer.add_forwarding_rule(remote_route, local_route)
+    main thread->>TransportForwarder - Bar: launch TransportForwarder - Bar
+    activate TransportForwarder - Bar
+    main thread->>UPClientBar owned thread / task: (within ustreamer.add_forwarding_rule()) <br> remote_route.transport.lock().await.register_listener <br> (uauthority_to_uuri(local_route.authority), forwarding_listener).await
+    activate UPClientBar owned thread / task
+
+    loop Park the main thread, let background tasks run until closing UStreamer app
+
+        par UPClientFoo thread / task calls ForwardingListener.on_receive()
+
+            UPClientFoo owned thread / task->>UPClientFoo owned thread / task: forwarding_listener.on_receive(UMessage)
+            UPClientFoo owned thread / task-->>TransportForwarder - Foo: Send UMesssage over channel
+            TransportForwarder - Foo->>TransportForwarder - Foo: out_transport.send(UMessage) <br> (out_transport => utransport_bar in this case)
+
+        end
+
+        par UPClientBar thread / task calls ForwardingListener.on_receive()
+
+            UPClientBar owned thread / task->>UPClientBar owned thread / task: forwarding_listener.on_receive(UMessage)
+            UPClientBar owned thread / task-->>TransportForwarder - Bar: Send UMesssage over channel
+            TransportForwarder - Bar->>TransportForwarder - Bar: out_transport.send(UMessage) <br> (out_transport => utransport_foo in this case)
+
+        end
+
+        deactivate UPClientFoo owned thread / task
+        deactivate UPClientBar owned thread / task
+        deactivate TransportForwarder - Foo
+        deactivate TransportForwarder - Bar
+
+    end
+```
+
+### Generating cargo docs locally
+
+Documentation can be generated locally with:
+```bash
+cargo doc --package up-streamer --open
+```
+which will open your browser to view the docs.
+
+## Getting Started
+
+### Working with the library
+
+`up-streamer-rust` is generic and pluggable and can serve your needs so long as
+1. Each transport you want to bridge over has a `up-client-foo-rust` library
+   and UPClientFoo struct which has `impl`ed `UTransport`
+2. `UTransportBuilder` has been `impl`ed on a struct so that a 
+   `Box<dyn UTransport>` can be safely created by the `UTransportRouter`
+   in the proper thread's context
+
+### Usage
+
+After following along with the [cargo docs](#generating-cargo-docs-locally) generated to add all your forwarding routes, you'll then need to keep the instantiated `UStreamer`, `UTransportRouter`, and `UTransportRouterHandle` around and then pause the main thread, so it will not exit, while the routing happens in the background threads spun up.
+
+## Implementation Status
+
+- [x] Routing of Request, Response, and Notification Messages
+- [ ] Routing of Publish messages (requires further development of uSubscription interface)
```

Comment:
I just noticed that uStreamer should handle Req/Resp/Notification and Publish in a different manner. The former can be listened to by destination UAuthority, while the latter is listened to by origin UAuthority. I should adjust the test in up-client-zenoh-rust.